src/common/metrics/ngx_wa_histogram.c

```diff
@@ -160,13 +178,23 @@ histogram_log(ngx_wa_metrics_t *metrics, ngx_wa_metric_t *m, uint32_t mid)
 
 
 ngx_int_t
-ngx_wa_metrics_histogram_add_locked(ngx_wa_metrics_t *metrics,
-    ngx_wa_metric_t *m)
+ngx_wa_metrics_histogram_add_locked(ngx_wa_metrics_t *metrics, uint32_t *bins,
+    uint16_t cn_bins, ngx_wa_metric_t *m)
 {
     size_t                        i;
-    static uint16_t               n_bins = NGX_WA_METRICS_BINS_INIT;
+    uint16_t                      j = 0, n_bins = NGX_WA_METRICS_BINS_INIT;
```

Comment:
`j` should be declared as per `i` (type and line) since it's a loop control variable.